sympy/core/tests/test_evalf.py

```diff
@@ -579,3 +579,16 @@ def test_issue_13425():
 
 def test_issue_17421():
     assert N(acos(-I + acosh(cosh(cosh(1) + I)))) == 1.0*I
+
+
+def test_issue_20291():
+    from sympy import FiniteSet, Complement, Intersection, Reals, EmptySet
+    a = Symbol('a')
+    b = Symbol('b')
+    A = FiniteSet(a, b)
+    assert A.evalf(subs={a: 1, b: 2}) == FiniteSet(1, 2)
+    B = FiniteSet(a-b, 1)
+    assert B.evalf(subs={a: 1, b: 2}) == FiniteSet(-1, 1)
```

Comment:
I think that the subs arguments should be evalfed to give Float e.g.:
```
In [9]: (x**y).evalf(subs={y:1, x:1})
Out[9]: 1.00000000000000
```
Is that not what happens with other evalf substitutions?

Reply:
It does return a float when run but it's not a problem. I was following the design of this:
https://github.com/sympy/sympy/blob/e036b1cfa1157b796b19ed0a1a0af97d40269366/sympy/core/tests/test_evalf.py#L505-L510

Reply:
Oh, okay. Well I think if the output is a Float then the test should show a Float. Ideally Float and Integer wouldn't compare equal like this (see #20033 where it looks like I had to change this test).